pyomo/dataportal/factory.py

```diff
@@ -15,58 +15,14 @@
 
 import logging
 import pyutilib.misc
-from pyomo.common.plugin import (
-    implements, Interface, Plugin, PluginFactory, CreatePluginFactory,
-    PluginError, ExtensionPoint )
+from pyomo.common.factory import Factory
 
 logger = logging.getLogger('pyomo.core')
 
 
-class IDataManager(Interface):
-
-    def available(self):
-        """ Returns True if the data manager can be executed """
-        pass
-
-    def requirements(self):
-        """ Return a string describing the packages that need to be installed for this plugin to be available """
-        pass
-
-    def initialize(self, filename, **kwds):
-        """ Prepare to read a data file. """
-        pass
-
-    def add_options(self, **kwds):
-        """ Add options """
-        pass
-
-    def open(self):
-        """ Open the data file. """
-        pass
-
-    def close(self):
-        """ Close the data file. """
-        pass
-
-    def read(self):
-        """ Read the data file. """
-        pass
-
-    def process(self, model, data, default):
-        """ Process the data. """
-        pass
-
-    def clear(self):
-        """ Reset Plugin. """
-        pass
-
-
-class UnknownDataManager(Plugin):
-
-    implements(IDataManager)
+class UnknownDataManager(object):
```

Comment:
Why would UnknownDataManager to be registered?  It's never created by the user, which is the point of a Factory.  The data manager logic knows about this class directly, and it can create it as appropriate.

Note that different Factory subclasses may act differently w.r.t. handling of "unknown" factory elements.  I think this is reasonable.  While the default behavior of Factory is to return None, which isn't very helpful for users.  But returning UnknownFactoryObject is just as vague.

Moreover, it's hard to standardize the return value given that different factories override the __call__ logic.